great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py

```diff
@@ -131,11 +132,14 @@ def _build_parameters(
         unique_values: Set[Any] = _get_unique_values_from_nested_collection_of_sets(
             collection=metric_values
         )
+
         unique_values_as_array: np.ndarray = np.asarray(unique_values)
         if unique_values_as_array.ndim == 0:
             unique_values_as_array = np.asarray([unique_values])
 
-        details["parse_strings_as_datetimes"] = is_ndarray_datetime_dtype(
+        details["parse_strings_as_datetimes"] = datetime_semantic_domain_type(
```

Comment:
Is this now an overloaded term? It seems like we might want a different name due to the deprecated Expectation parameter.

Reply:
@NathanFarmer This term was introduced in a previous PR -- intentionally overloaded, so that if we ever try to fix/change it, we can do it consistently throughout.  Thanks.

Reply:
I'm still confused about why we need to set `parse_strings_as_datetimes` at all. If this is deprecated, we should avoid using it right?

Reply:
@NathanFarmer It is just the "code word" for now (from previous PR).  What is changed in this PR is that the right-hand side is computed more efficiently.  Thanks.